### PR TITLE
Add "get_all_pages" function to all endpoints

### PR DESCRIPTION
**Pull request by ChaseParate**
This is a more convenient way to get all of the resources from a given endpoint.

### CODE REVIEW

**Comment by mlemesle** [comment]
Hey @ChaseParate !

Do you plan to do the changes or should I take care of it ? I'd like to make a release soon :slightly_smiling_face: 

Please let me know, thanks

File: src/endpoint.rs [hunk 1]
```diff
@@ -38,6 +38,34 @@ macro_rules! endpoint {
             rustemon_client.get_by_url::<NamedApiResourceList<$type>>(url).await
         }
 
+        /// Returns all pages from the given resource.
+        ///
+        /// # Arguments
+        ///
+        /// `rustemon_client` - The [RustemonClient] to use to access the resource.
+        pub async fn get_all_pages(rustemon_client: &RustemonClient) -> Result<Vec<NamedApiResource<$type>>, Error> {
```
Comment (mlemesle) [inline src/endpoint.rs:46]: This can be reworked in order to use the `next` field in `NamedApiResourceList`

**Review by mlemesle — CHANGES_REQUESTED** [review]

**Comment by ChaseParate** [comment]
Sorry, I've been super busy! I've just implemented the changes you mentioned and it seems much more concise! I was wondering if the function should be named "get_all" or "get_all_resources" instead of "get_all_pages". What do you think?

**Comment by ChaseParate** [comment]
Also, perhaps there is a way to return the paginated data without needing to await every page? Something like a [Stream](https://docs.rs/futures/latest/futures/stream/trait.Stream.html) comes to mind, but I'm not too familiar with the `futures` crate.

**Review by ChaseParate — COMMENTED** [review]
This was a mistake: I just synced up my fork to the main repo.

**Comment by mlemesle** [comment]
Hey there, no worries !

Thanks for the update, this is much more readable.

Regarding `futures` to avoid awaiting every pages, do you wanna do it ? Please let me know :slightly_smiling_face:

**Comment by ChaseParate** [comment]
I did a few prototypes with `Stream`s, but I just couldn't quite get it to work. Maybe it would be worthwhile to investigate later.

**Review by mlemesle — APPROVED** [review]

**Comment by mlemesle** [comment]
Thanks for your contribution !

**Pull request merged**